cmd/kubeapps-apis/server/server.go

```diff
@@ -34,16 +35,34 @@ import (
 	pluginsGRPCv1alpha1 "github.com/kubeapps/kubeapps/cmd/kubeapps-apis/gen/core/plugins/v1alpha1"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/reflection"
+	"google.golang.org/grpc/status"
 	"google.golang.org/protobuf/encoding/protojson"
 	log "k8s.io/klog/v2"
 )
 
+// LogRequest is a gRPC UnaryServerInterceptor that will log the API call
+func LogRequest(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (response interface{}, err error) {
+	start := time.Now()
+	res, err := handler(ctx, req)
+
+	// Include micro seconds in timestamp
+	// Format string : [timestamp] [status code] [duration] [full path]
+	// 2021-11-29 15:10:21.642313 OK 97.752µs /kubeappsapis.core.packages.v1alpha1.PackagesService/GetAvailablePackageSummaries
+
+	fmt.Printf("%s %v %s %s\n", time.Now().Format("2006-01-02 15:04:05.000000"),
+		status.Code(err),
+		time.Since(start),
+		info.FullMethod)
+
+	return res, err
+}
+
 // Serve is the root command that is run when no other sub-commands are present.
 // It runs the gRPC service, registering the configured plugins.
 func Serve(serveOpts core.ServeOptions) error {
 	// Create the grpc server and register the reflection server (for now, useful for discovery
 	// using grpcurl) or similar.
-	grpcSrv := grpc.NewServer()
+	grpcSrv := grpc.NewServer(grpc.ChainUnaryInterceptor(LogRequest))
```

Comment:
Ideally we'd create the logger here (`logger := log.New(...)`), but to include it in the `LogRequest` function (without instantiating it every time), we'll probably need to use a closure. The simplest way would be to just create the logger here, then move the definition of your LogRequest function inline here so it can reference the logger. That way the logger is only created once. 

That would be fine and do the job, but if you're more familiar with closures, you could instead create a function that takes the created logger as input and returns the request logger function, `CreateRequestLogger(log.Logger) func(context.Context, ...) (interface{}, error) { ... }`, which is a bit nicer, if you're used to closures in Go. Fine either way.